aws-lc-rs/src/rsa/encoding.rs

```diff
@@ -130,14 +130,19 @@ pub(in crate::rsa) mod rfc5280 {
         encoding::PublicKeyX509Der,
         error::{KeyRejected, Unspecified},
         ptr::LcPtr,
-        rsa::key::is_rsa_key,
+        rsa::key::{is_rsa_key, key_size_bytes},
     };
     use aws_lc::{EVP_marshal_public_key, EVP_parse_public_key, EVP_PKEY};
 
     pub(in crate::rsa) fn encode_public_key_der(
         key: &LcPtr<EVP_PKEY>,
     ) -> Result<PublicKeyX509Der<'static>, Unspecified> {
-        let mut der = LcCBB::new(1024);
+        // Data shows that the SubjectPublicKeyInfo is roughly 356% to 375% increase in size comapred to the RSA key
+        // size in bytes for keys ranging from 2048-bit to 4096-bit. So size the initial capacity to be roughly
+        // 400% as a consernative estimate to avoid needing to reallocate for any key in that range.
+        let key_size_bytes = key_size_bytes(key);
+
+        let mut der = LcCBB::new(key_size_bytes + (key_size_bytes * 4));
```

Comment:
NP: this is just 5 times `key_size_bytes` -- but maybe you broke it out for clarity regarding the comment above? (Also a couple of spelling mistakes in the comment.)
```
let mut der = LcCBB::new(5 * key_size_bytes);
```

Reply:
Really need to look for a spell check extension for vscode 😬.

Yeah I have it broken out just to make it clear, but I can optimize it to your suggestion and leave a comment.